### DISCUSSION TITLE
Device Incorrectly Detected - How to Fix?

### DISCUSSION THREAD

**Discussion by jjwork** [Koenkk/zigbee2mqtt#12250] [Devices]
Hello !

I have a few of these [https://www.zigbee2mqtt.io/devices/NAS-PD07.html](url)

However, they are all being recognised as these [https://www.zigbee2mqtt.io/devices/TS0202.html#tuya-ts0202](url)

Can I please get some guidance on how to fix this ?

All the best

Jason

**Comment by Koenkk** [comment]
Can you provide the data/database.db entry of your device?

**Reply by jjwork** [reply]
Hi Koenkk !

Thanks so much for the reply !

I am so sorry to trouble you, but I do not know where to go to get you that information.

Can you point me to a link to show me the steps ?

Apologies for the inconveneince

Jason

**Reply by Koenkk** [reply]
Or a screenshot of the device in the frontend. e.g.: 

<img width="859" alt="Screenshot 2022-04-29 at 22 15 39" src="https://user-images.githubusercontent.com/2892853/166063244-c904ba82-9097-4cc7-bbc3-5c5dae746d2f.png">

**Reply by jjwork** [reply]
Yup, that I can do !  

![image](https://user-images.githubusercontent.com/5220788/166142996-2074893b-5b5c-46ea-8c6e-48044df52888.png)

**Reply by Koenkk** [reply]
Could you check if it works correctly now?

Changes will be available in the dev branch in a few hours from now. (https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html)

**Reply by jjwork** [reply]
Oh man - you are the best !!!!

Unfortunately I don't have a dev environment to test :-(

I'll wait patiently till it makes its way over to release build in HASS.

Thank you so much for the super quick response and I very much appreciate your support !

**Reply by jjwork** [reply]
Hi Koenkk,

I finally got around to doing a re-build of my HASS environment and using the EDGE repository.

Works great - it recognizes the devices correctly !!

Thank yo so much for your super help !!

Jason

**Reply by jjwork** [reply]
Hi Koenkk !

Sorry to trouble you ... is this fix scheduled to make its way to the release builds ?

All the best and thank you again !

Jason

**Reply by Koenkk** [reply]
It is in 1.27.0
